tensorflow/lite/testing/generate_examples_lib.py

```diff
@@ -266,14 +266,20 @@ def write_test_cases(fp, model_name, examples):
 }
 
 
-def create_tensor_data(dtype, shape, min_value=-100, max_value=100):
+def create_tensor_data(dtype, shape, min_value=-100, max_value=100, include_nan=False):
   """Build tensor data spreading the range [min_value, max_value)."""
 
   if dtype in _TF_TYPE_INFO:
     dtype = _TF_TYPE_INFO[dtype][0]
 
   if dtype in (tf.float32, tf.float16):
     value = (max_value-min_value)*np.random.random_sample(shape)+min_value
+    if include_nan:
+        mask=np.zeros(shape, dtype=bool)
+        mask[:value.size//2] = True
```

Comment:
are you sure this is correct? when it's >1 dimensions, seems this will be a broadcast operation.

and also can you format it as well?

Reply:
@renjie-liu , I have updated the code and replaced the logic for > 1 dimension as well, kindly have a look.

Regards
Amit